eprihlaska/validators.py

```diff
@@ -32,3 +32,21 @@ def __call__(self, form, field):
         birth_no = int(mo.group(1) + mo.group(2))
         if birth_no % 11 != 0:
             raise validators.ValidationError(self.message + ' ' + self.mod_err)
+
+
+class DateValidator:
+    def __init__(self, date=None, message=None):
+        self.date = date
+
+        if not message:
+            message = 'Správny formát je DD.MM.RRRR'
+        self.message = message
+
+    def __call__(self, form, field):
+        date_reg = re.compile(r'^(?:(?:31(\.)(?:0?[13578]|1[02]))\1|(?:(?:29|30)(\.)(?:0?[1,3-9]|1[0-2])\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:29(\.)(?:0?2)\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\d|2[0-8])(\.)(?:(?:0?[1-9])|(?:1[0-2]))\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$')
```

Comment:
@JakubNvk Toto je hrozne komplikovany regex. Slo by ho trochu zjednodusit? 

Myslim, ze by nam uplne stacilo `r'^(\d\d)\.(\d\d)\.(\d\d\d\d)$'` a potom vyskusat, ci je ten datum validny, napriklad nejak [takto](https://stackoverflow.com/questions/16870663/how-do-i-validate-a-date-string-format-in-python). Ci sa to takto zda byt lepsie?

Este by som odporucal tento kompilovany regex dat do `__init__`-u, aby sa to nekompilovalo znova a znova.

Vdaka!

Reply:
Cokolvek je lepsie ako henta abominacia...prerobim to :)